### DISCUSSION TITLE
Flows: is it possible to run a graphql query?

### DISCUSSION THREAD

**Discussion by DeanKamali** [directus/directus#15214] [Archive]
Hello everyone, 

I'm trying to run a GraphQL query via the webhook & POST method available in flows, but I haven't been successful. 

I saw there is a new "Run script" option which is quite exciting, and thought I could use it to run GraphQL queries,  but I'm not sure how to install/import directus SDK or other packages in the Standard cloud offering. 

 I tried converting my query to JSON like so: 

`{
  "query": "query {	revisions_by_id(id: 22414) { id data delta	}}"
}`

But keep getting bad requests.

**Comment by DeanKamali** [comment]
I managed to get it to work by removing spaces from my query string like so:
```
{
  "query": "query{revisions_by_id(id: 22414){id data delta}}"
}
```

It would be nice to have a native implementation so that we can enter the query and It will be converted into a JSON string like the example above.

**Comment by rijkvanzanten** [comment]
Sweet! That being said, I'd strongly recommend using the "Read" operation rather than the "Script" operation + a GraphQL query, as the former will use the internal services to go straight to the database, saving you a network request (and the latency/additional events that come with that)

**Reply by nickreese** [reply]
After spending several hours on this, I can't seem to sort out how to do a query against `directus_revisions` from a flow. 

<img width="408" alt="image" src="https://user-images.githubusercontent.com/3580879/202910510-745c5ef1-ff48-4bcb-acd9-151de9b0d803.png">

Any tips? I can't seem to get the `\t` to disappear from the collection key.

**Reply by nickreese** [reply]
If anyone else goes this route (changing slugs on a revision, look at a hook instead)

**Reply by luciandavidescu** [reply]
@nickreese this is exactly what I'm looking for. In a data read on directus_revisions:
```
{
    "filter": {
        "item": {
            "_eq": "{{$trigger.keys}}"
        }
    },
    "limit": "1",
    "offset": "1",
    "sort": [
        "-activity"
    ]
}
```
THis gets the last but one revision and from there you can get the previous slug with something like `{{$last[0].data.slug}}`

**Reply by toby-reid-rocketlab** [reply]
> After spending several hours on this, I can't seem to sort out how to do a query against `directus_revisions` from a flow.
> 
> <img alt="image" width="408" src="https://user-images.githubusercontent.com/3580879/202910510-745c5ef1-ff48-4bcb-acd9-151de9b0d803.png">
> 
> Any tips? I can't seem to get the `\t` to disappear from the collection key.

@nickreese to get rid of the `\t` use the raw input by clicking the `{}` next to the Collection title above the dropdown. Then make sure all text and characters are removed (it should show a placeholder `{{$trigger.payload.example}}`). Then type out the system collection and save.
